state/utils/utils.go

```diff
@@ -13,6 +13,26 @@ limitations under the License.
 
 package utils
 
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+func JSONStringify(value any) ([]byte, error) {
+	switch value := value.(type) {
+	case []byte:
+		return value, nil
+	case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64, float32, float64, bool:
+		return []byte(fmt.Sprintf("%v", value)), nil
```

Comment:
Maybe a bit of a nit, but I think this would be better (better perf) if each type were handled separately, such as:

```go
case int, int8, int16, int32, int64:
    return []byte(strconv.FormatInt(int64(value), 10))
case uint, uint8, uint16, uint32, uint64:
    return []byte(strconv.FormatUint(uint64(value), 10))
case bool:
    if value {
        return []byte("true")
    } else {
        return []byte("false")
    }
// Etc
```